tests/event/event.cpp

```diff
@@ -0,0 +1,466 @@
+// the async handler associated with the context" means when two queues override
+// their (shared) context's async handler.
+TEST_CASE(
+    "event::wait_and_throw reports asynchronous errors from related events on "
+    "other queues",
+    "[event][todo-spec][!mayfail]") {
+  test_exception_handler teh1;
+  test_exception_handler teh2;
+  REQUIRE(teh1.get_queue() != teh2.get_queue());
+
+  auto e1 = make_throwing_host_event(teh1.get_queue(), "some-error");
+  auto e2 = make_throwing_host_event(teh2.get_queue(), "another-error", {e1});
+
+  SECTION("event::wait_and_throw") { e2.wait_and_throw(); }
+  SECTION("event::wait_and_throw(std::vector<event>)") {
+    sycl::event::wait_and_throw(std::vector{e2});
+  }
+
+  CHECK(teh2.count() == 2);
+  CHECK(teh2.has("some-error"));
+  CHECK(teh2.has("another-error"));
+
+  // This should be a no-op as the error already has been consumed
+  e1.wait_and_throw();
+  CHECK(teh1.count() == 0);
+}
+
+TEST_CASE("event::wait_and_throw only reports unconsumed asynchronous errors",
+          "[event]") {
+  test_exception_handler teh;
+  make_throwing_host_event(teh.get_queue(), "some-error").wait_and_throw();
+  teh.clear();
+
+  auto e = make_throwing_host_event(teh.get_queue(), "another-error");
+
+  SECTION("event::wait_and_throw") { e.wait_and_throw(); }
+  SECTION("event::wait_and_throw(std::vector<event>)") {
+    sycl::event::wait_and_throw(std::vector{e});
+  }
+
+  CHECK(teh.count() == 1);
+  CHECK(teh.has("another-error"));
+}
+
+TEST_CASE("event::get_info returns correct command execution status",
+          "[event]") {
+  // First check that return value is of expected type
+  check_get_info_param<sycl::info::event, sycl::info::event_command_status,
+                       sycl::info::event::command_execution_status>(
+      make_device_event());
+
+  SECTION("for host_task event") {
+    resolvable_host_event rhe;
+    auto& event = rhe.get_sycl_event();
+
+    const auto status1 =
+        event.get_info<sycl::info::event::command_execution_status>();
+    CHECK((status1 == sycl::info::event_command_status::submitted ||
+           status1 == sycl::info::event_command_status::running));
+
+    rhe.resolve();
+    event.wait();
+
+    const auto status2 =
+        event.get_info<sycl::info::event::command_execution_status>();
+    CHECK(status2 == sycl::info::event_command_status::complete);
+  }
+
+  SECTION("for device kernel event") {
+    // Here we cannot really control the execution status,
+    // so we cannot be as thorough
+    auto e1 = make_device_event();
+    auto e2 = make_device_event({e1});
+    e2.wait();
+    // Since e2's CGF depends on e1, the latter must have completed by now
+    const auto status =
+        e1.get_info<sycl::info::event::command_execution_status>();
+    CHECK(status == sycl::info::event_command_status::complete);
+  }
+}
+
+// TODO: Figure out if/how we want to test this.
+// => Must throw exception w/ ercc:backend_mismatch if querying a paramter for
```

Comment:
```suggestion
// => Must throw exception w/ errc::backend_mismatch if querying a parameter for
```

Reply:
Fixed

Reply:
You seem to miss fixes for 2 typos out of 3.

Reply:
Woops, you're right. I blame my spellchecker who seems to think that "paramter" is a word...